src/Data/Nat/DivMod/Core.agda

```diff
@@ -0,0 +1,98 @@
+  modₕ (suc a + acc) n b 0       ∎
+
+modₕ-skipToEnd : ∀ acc n a b → modₕ acc n a (a + b) ≡ acc + a
+modₕ-skipToEnd acc n zero    b = sym (+-identityʳ acc)
+modₕ-skipToEnd acc n (suc a) b = begin
+  modₕ (suc acc) n a (a + b) ≡⟨ modₕ-skipToEnd (suc acc) n a b ⟩
+  suc acc + a                ≡⟨ sym (+-suc acc a) ⟩
+  acc + suc a                ∎
+
+a[modₕ]1≡0 : ∀ a → modₕ 0 0 a 0 ≡ 0
+a[modₕ]1≡0 zero    = refl
+a[modₕ]1≡0 (suc a) = a[modₕ]1≡0 a
+
+n[modₕ]n≡0 : ∀ acc v → modₕ acc (acc + v) (suc v) v ≡ 0
+n[modₕ]n≡0 acc v = modₕ-skipTo0 acc (acc + v) v 1
+
+mod-lemma : ∀ acc d n → modₕ acc (acc + n) d n ≤ acc + n
+mod-lemma acc zero    n       = m≤m+n acc n
+mod-lemma acc (suc d) zero    = mod-lemma zero d (acc + 0)
+mod-lemma acc (suc d) (suc n)
+          rewrite +-suc acc n = mod-lemma (suc acc) d n
+
+modₕ-absorb : ∀ acc a n → modₕ 0 (acc + n) (modₕ acc (acc + n) a n) (acc + n) ≡ modₕ acc (acc + n) a n
+modₕ-absorb acc zero    n       = modₕ-skipToEnd 0 (acc + n) acc n
+modₕ-absorb acc (suc a) zero    rewrite +-identityʳ acc = modₕ-absorb 0 a acc
+modₕ-absorb acc (suc a) (suc n) rewrite +-suc acc n = modₕ-absorb (suc acc) a n
+
+a+n[modₕ]n≡a[modₕ]n : ∀ acc a n → modₕ acc (acc + n) (acc + a + suc n) n ≡ modₕ acc (acc + n) a n
+a+n[modₕ]n≡a[modₕ]n acc zero n rewrite +-identityʳ acc = begin
+  modₕ acc (acc + n) (acc + suc n) n   ≡⟨ cong (λ v → modₕ acc (acc + n) v n) (+-suc acc n) ⟩
+  modₕ acc (acc + n) (suc acc + n) n   ≡⟨ modₕ-skipTo0 acc (acc + n) n (suc acc) ⟩
+  modₕ (acc + n) (acc + n) (suc acc) 0 ≡⟨⟩
+  modₕ 0 (acc + n) acc (acc + n)       ≡⟨ modₕ-skipToEnd 0 (acc + n) acc n ⟩
+  acc                                  ∎
+a+n[modₕ]n≡a[modₕ]n acc (suc a) zero rewrite +-identityʳ acc = begin
+  modₕ acc acc (acc + suc a + 1)   0   ≡⟨ cong (λ v → modₕ acc acc v 0) (+-comm (acc + suc a) 1) ⟩
+  modₕ acc acc (1 + (acc + suc a)) 0   ≡⟨⟩
+  modₕ 0   acc (acc + suc a)       acc ≡⟨ cong (λ v → modₕ 0 acc v acc) (+-comm acc (suc a)) ⟩
+  modₕ 0   acc (suc a + acc)       acc ≡⟨ cong (λ v → modₕ 0 acc v acc) (sym (+-suc a acc)) ⟩
+  modₕ 0   acc (a + suc acc)       acc ≡⟨ a+n[modₕ]n≡a[modₕ]n 0 a acc ⟩
+  modₕ 0   acc a                   acc ∎
+a+n[modₕ]n≡a[modₕ]n acc (suc a) (suc n) rewrite +-suc acc n = begin
+  modₕ acc       (suc acc + n) (acc + suc a + suc (suc n)) (suc n) ≡⟨ cong (λ v → modₕ acc (suc acc + n) (v + suc (suc n)) (suc n)) (+-suc acc a) ⟩
+  modₕ acc       (suc acc + n) (suc acc + a + suc (suc n)) (suc n) ≡⟨⟩
+  modₕ (suc acc) (suc acc + n) (acc + a + (suc (suc n)))   n       ≡⟨ cong (λ v → modₕ (suc acc) (suc acc + n) v n) (sym (+-assoc (acc + a) 1 (suc n))) ⟩
+  modₕ (suc acc) (suc acc + n) (acc + a + 1 + suc n)       n       ≡⟨ cong (λ v → modₕ (suc acc) (suc acc + n) (v + suc n) n) (+-comm (acc + a) 1) ⟩
+  modₕ (suc acc) (suc acc + n) (suc acc + a + suc n)       n       ≡⟨ a+n[modₕ]n≡a[modₕ]n (suc acc) a n ⟩
+  modₕ (suc acc) (suc acc + n) a                           n       ∎
+
```

Comment:
Probably worth naming `modₕ acc (suc acc + n)` and `modₕ (suc acc) (suc acc + n)` here.
They are used in the expressions and the `cong`s!